### PR TITLE
Enable PWM on STM32WB

### PR DESCRIPTION
**Pull request by rickconrey**
Enables PWM and Timers 1, 2, 16 and 17 on STM32WB

### CODE REVIEW

File: soc/arm/st_stm32/stm32wb/dts_fixup.h [hunk 1]
```diff
@@ -103,4 +103,28 @@
 #define DT_I2C_3_CLOCK_BITS			DT_ST_STM32_I2C_V2_40005C00_CLOCK_BITS
 #define DT_I2C_3_CLOCK_BUS			DT_ST_STM32_I2C_V2_40005C00_CLOCK_BUS
 
+#define DT_PWM_STM32_1_DEV_NAME	DT_ST_STM32_PWM_40012C00_PWM_LABEL
```
Comment (erwango) [inline soc/arm/st_stm32/stm32wb/dts_fixup.h]: Can you fix alignment?

File: soc/arm/st_stm32/stm32wb/dts_fixup.h [hunk 2]
```diff
@@ -103,4 +103,28 @@
...
+#define DT_PWM_STM32_1_PRESCALER	DT_ST_STM32_PWM_40012C00_PWM_ST_PRESCALER
+#define DT_TIM_STM32_1_BASE_ADDRESS	DT_ST_STM32_TIMERS_40012C00_BASE_ADDRESS
+#define DT_TIM_STM32_1_CLOCK_BITS	DT_ST_STM32_TIMERS_40012C00_CLOCK_BITS
+#define DT_TIM_STM32_1_CLOCK_BUS	DT_ST_STM32_TIMERS_40012C00_CLOCK_BUS
+
+#define DT_PWM_STM32_2_DEV_NAME	DT_ST_STM32_PWM_40000000_PWM_LABEL
```
Comment (erwango) [inline soc/arm/st_stm32/stm32wb/dts_fixup.h]: ditto

File: boards/arm/nucleo_wb55rg/pinmux.c [hunk 3]
```diff
@@ -30,6 +30,9 @@ static const struct pin_config pinconf[] = {
 	{STM32_PIN_PC0, STM32WBX_PINMUX_FUNC_PC0_I2C3_SCL},
 	{STM32_PIN_PC1, STM32WBX_PINMUX_FUNC_PC1_I2C3_SDA},
 #endif /* CONFIG_I2C_3 */
+#ifdef CONFIG_PWM_STM32_2
```
Comment (erwango) [inline boards/arm/nucleo_wb55rg/pinmux.c]: Please update board documentation as well.

**Review by erwango — CHANGES_REQUESTED** [review]
Few remarks otherwise looks fine.

File: boards/arm/nucleo_wb55rg/pinmux.c (hunk 3 above)
Comment (rickconrey) [reply boards/arm/nucleo_wb55rg/pinmux.c]: updated

**Review by dbkinder — APPROVED** [review]
+1 for doc changes

File: soc/arm/st_stm32/stm32wb/dts_fixup.h [hunk 4]
```diff
@@ -110,4 +110,28 @@
 #define DT_I2C_3_CLOCK_BITS			DT_ST_STM32_I2C_V2_40005C00_CLOCK_BITS
 #define DT_I2C_3_CLOCK_BUS			DT_ST_STM32_I2C_V2_40005C00_CLOCK_BUS
 
+#define DT_PWM_STM32_1_DEV_NAME		DT_ST_STM32_PWM_40012C00_PWM_LABEL
```
Comment (erwango) [inline soc/arm/st_stm32/stm32wb/dts_fixup.h]: Seems alignment is still wrong.

File: soc/arm/st_stm32/stm32wb/dts_fixup.h [hunk 5]
```diff
@@ -110,4 +110,28 @@
...
+#define DT_PWM_STM32_1_PRESCALER		DT_ST_STM32_PWM_40012C00_PWM_ST_PRESCALER
+#define DT_TIM_STM32_1_BASE_ADDRESS		DT_ST_STM32_TIMERS_40012C00_BASE_ADDRESS
+#define DT_TIM_STM32_1_CLOCK_BITS		DT_ST_STM32_TIMERS_40012C00_CLOCK_BITS
+#define DT_TIM_STM32_1_CLOCK_BUS		DT_ST_STM32_TIMERS_40012C00_CLOCK_BUS
+
+#define DT_PWM_STM32_2_DEV_NAME		DT_ST_STM32_PWM_40000000_PWM_LABEL
```
Comment (erwango) [inline soc/arm/st_stm32/stm32wb/dts_fixup.h]: ditto

**Review by erwango — CHANGES_REQUESTED** [review]

File: soc/arm/st_stm32/stm32wb/dts_fixup.h (hunk 4 above)
Comment (rickconrey) [reply soc/arm/st_stm32/stm32wb/dts_fixup.h]: hmmm.. how about now?

**Review by erwango — APPROVED** [review]

**Pull request merged**